custom_components/tahoma/siren.py

```diff
@@ -0,0 +1,114 @@
+"""Support for Overkiz sirens."""
+from homeassistant.components.siren import DOMAIN as SIREN, SirenEntity
+from homeassistant.components.siren.const import (
+    ATTR_DURATION,
+    SUPPORT_DURATION,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CORE_ON_OFF_STATE, DOMAIN
+from .entity import OverkizEntity
+
+COMMAND_MEMORIZED_VOLUME = "memorizedVolume"
+COMMAND_RING_WITH_SINGLE_SIMPLE_SEQUENCE = "ringWithSingleSimpleSequence"
+COMMAND_STANDARD = "standard"
+
+STATE_ON = "on"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+):
+    """Set up the Overkiz sirens from a config entry."""
+    data = hass.data[DOMAIN][entry.entry_id]
+    coordinator = data["coordinator"]
+
+    entities = [
+        OverkizSiren(device.deviceurl, coordinator)
+        for device in data["platforms"][SIREN]
+    ]
+
+    async_add_entities(entities)
+
+
+class OverkizSiren(OverkizEntity, SirenEntity):
+    """Representation an Overkiz Switch."""
+
+    _attr_supported_features = SUPPORT_TURN_OFF | SUPPORT_TURN_ON | SUPPORT_DURATION
+
+    @property
+    def is_on(self):
+        """Get whether the siren is in on state."""
+        return self.executor.select_state(CORE_ON_OFF_STATE) == STATE_ON
+
+    async def async_turn_on(self, **kwargs):
+        """Send the on command."""
+
+        if kwargs.get(ATTR_DURATION):
+            duration = kwargs.get(ATTR_DURATION)
+        else:
+            duration = 2 * 60  # 2 minutes
```

Comment:
Why this default value?

Reply:
Took it from here:     https://www.tahomalink.com/enduser-mobile-web/steer-html5-client/vendor/somfy/io/siren/const.js

Reply:
I mean ATTR_DURATION can be missing?

Reply:
Yes, it should be optional, but I would have to double check! :-)